pytorch_lightning/trainer/trainer.py

```diff
@@ -414,9 +416,12 @@ def __init__(
         # nvidia setup
         self.set_nvidia_flags(self.is_slurm_managing_tasks, self.data_parallel_device_ids)
 
+        # Backward compatibility, TODO: remove in v0.8.0
+        if show_progress_bar is not None:
+            warnings.warn("Argument `show_progress_bar` is now set by `progress_bar_refresh_rate` since v0.7.1"
+                          " and this method will be removed in v0.8.0", DeprecationWarning)
```

Comment:
let's add deprecated setter (in the mixin deprecated API) where the `show_progress_bar` as True sets `self.progress_bar_refresh_rate = 1` with the deprecation warning...
and here in the iff body have only
```suggestion
            self.show_progress_bar = show_progress_bar
```